packages/react-sdk/src/hooks/storage/useCustomUploader.ts

```diff
@@ -0,0 +1,83 @@
+import { useState, useCallback } from 'react';
+import { useArweaveStorage } from './arweave/useArweave';
+import { WalletContextState } from '@solana/wallet-adapter-react';
+import { Connection } from '@solana/web3.js';
+import { SessionWalletInterface } from '../session';
+
+export type UploaderType = 'arweave' | 'genesysgo' | 'custom';
+
+export interface UploaderFunction {
+  (data: string): Promise<string>;
+}
+
+export const useCustomUploader = (
+  uploaderType: UploaderType,
+  wallet: WalletContextState,
+  connection: Connection,
+  cluster: "devnet" | "mainnet-beta",
+  useSession: boolean,
+  session?: SessionWalletInterface,
+  customUploader?: UploaderFunction,
+) => {
+  if (uploaderType === 'custom' && !customUploader) {
+    throw new Error('Custom uploader function is required for custom uploader type');
+  }
+
+  const arweaveStorage = useArweaveStorage(wallet, connection, cluster, useSession, session);
+
+  const getFileUploader = (): UploaderFunction => {
+    if (uploaderType === 'custom' && customUploader) {
```

Comment:
What happens here when the custom uploader needs access to the wallet or keypair to sign a transaction/message?